nautobot/core/api/views.py

```diff
@@ -721,14 +721,13 @@ class GetMenuAPIView(NautobotAPIVersionMixin, APIView):
 
     permission_classes = [IsAuthenticated]
 
-    def format_and_remove_hidden_menu(self, request, data, hide_restricted_ui):
+    def format_and_remove_hidden_menu(self, request, data, hide_restricted_ui=False):
         """
         Formats the menu data and removes hidden menu items based on user permissions.
 
         Args:
             request (HttpRequest): The request object.
             data (Union[dict, str]): The menu data to format and filter. Can be either a dictionary or a string.
-            hide_restricted_ui (bool): Flag indicating whether to hide restricted menu items.
```

Comment:
Let's discuss in standup - I don't know if we'd been thinking about the role of `HIDE_RESTRICTED_UI` in selectively disabling individual items based on user permissions; the focus of #4787 was really more around removing the ability for unauthenticated users to access the UI at all. We might want to keep `HIDE_RESTRICTED_UI` around but narrow its scope to only hiding/disabling UI elements for authenticated users based on their permissions?